java/src/main/java/org/hyperledger/fabric/client/FileCheckpointer.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+
+import javax.json.JsonObject;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.OpenOption;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+import javax.json.Json;
+import javax.json.JsonReader;
+import javax.json.JsonWriter;
+
+
+/**
+ * FileCheckpointer to store checkpointer state during file read write operations.
+ */
+public final class FileCheckpointer implements Checkpointer {
+
+    private long blockNumber;
+    private String transactionId = "";
+    private String path = "";
+    private final Reader fileReader;
+    private final Writer fileWriter;
+    private static final Set<OpenOption> OPEN_OPTIONS = Collections.unmodifiableSet(EnumSet.of(
+            StandardOpenOption.CREATE,
+            StandardOpenOption.READ,
+            StandardOpenOption.WRITE
+    ));
+    private final FileChannel fileChannel;
+    private static final String CONFIG_KEY_BLOCK = "blockNumber";
+    private static final String CONFIG_KEY_TRANSACTIONID = "transactionId";
+
+    FileCheckpointer(final String path) throws IOException {
```

Comment:
The constructor should take a [Path](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html) instead of a String parameter.